File: 1.CPU_Memory_Disk_Network/Network/Network_Programming_In_Practice/NPIP_Go/t01_tcp_handshake/server/TcpHandshake1Server.go
package main

import (
	"fmt"
	"net"
)

const HOST = "localhost"
const PORT = 9123

// This server accept only one client connection intentionally.
func main() {
	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%d", HOST, PORT))
	defer listener.Close()
	panicIfError(err, "Error listening on port", PORT)
	fmt.Printf("Server is listening on port %d ...\n", PORT)
	conn, err := listener.Accept()
	defer conn.Close()
	panicIfError(err, "Error accepting connection")
	fmt.Println("A client connected.")
}

func panicIfError(err error, messages ...any) {
	if err != nil {
		panic(fmt.Sprint(messages, "\nerror: ", err))
	}
}

File: 1.CPU_Memory_Disk_Network/Network/Network_Programming_In_Practice/NPIP_Java/src/main/java/test/t01_tcp_handshake/TcpHandshake1Server.java
package test.t01_tcp_handshake;

import static java.lang.System.out;

import java.net.ServerSocket;
import java.net.Socket;

/**
 * TCP Handshake Server. This server accept only one client connection intentionally.
 *
 * @author Camio1945
 */
public class TcpHandshake1Server {
  protected static final int PORT = 9123;

  public static void main(String[] args) throws Exception {
    try (ServerSocket serverSocket = new ServerSocket(PORT)) {
      out.printf("\nServer is listening on port %d ...\n", PORT);
      Socket acceptedSocket = serverSocket.accept();
      if (acceptedSocket != null) {
        out.println("A client connected.");
      }
    }
  }
}
